include/drivers/flash.h

```diff
@@ -166,9 +163,6 @@ static inline int z_impl_flash_read(const struct device *dev, off_t offset,
  *  Write size and offset must be multiples of the minimum write block size
  *  supported by the driver.
  *
- *  Prior to the invocation of this API, the flash_write_protection_set needs
```

Comment:
This shouldn't just be dropped; it should be replaced with text that specifies that any necessary write protection management must be performed by the driver, with the driver responsible for ensuring the "write-protect" after the operation completes (successfully or not) matches the write-protect state when the operation was started.

Otherwise there isn't enough detail here to tell an implementer what this has to do.